.github/workflows/tests.yml

```diff
@@ -110,3 +110,20 @@ jobs:
         fi
         twine check dist/*
         twine upload dist/*
+
+    deploy-github-release:
+      # only deploy on tags, see https://stackoverflow.com/a/58478262/1320237
+      if: startsWith(github.ref, 'refs/tags/v')
+      needs:
+        - run-tests
+      runs-on: ubuntu-latest
+      environment:
+        name: github-release
+      steps:
+        - uses: actions/checkout@v2
+        - name: create release
+          uses: elgohr/Github-Release-Action@v4
+          env:
+            GITHUB_TOKEN: ${{ secrets.RELEASE_TOKEN }}
```

Comment:
Hm. This looks ok.
You will also need to create the environment with the secret here: https://github.com/collective/icalendar/settings/secrets/actions

I would personally also add a link to the documentation e.g. where you copied bits from so that we can tweak it more easily.

Once you created the environment, you can merge it. We will see how it does on the next release.